app/code/Bulbulatory/Recomendations/Ui/Component/Listing/Column/RecommenderEmail.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Bulbulatory\Recomendations\Ui\Component\Listing\Column;
+
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Ui\Component\Listing\Columns\Column;
+use \Magento\Customer\Api\CustomerRepositoryInterface;
+
+class RecommenderEmail extends Column
+{
+    protected $customerRepositoryInterface;
+
+    public function __construct(
+        CustomerRepositoryInterface $customerRepositoryInterface,
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        array $components = [],
+        array $data = []
+    ) {
+        parent::__construct($context, $uiComponentFactory, $components, $data);
+        $this->_customerRepositoryInterface = $customerRepositoryInterface;
+    }
+
+public function prepareDataSource(array $dataSource)
+{
+    if (isset($dataSource['data']['items'])) {
+        foreach ($dataSource['data']['items'] as &$item) {
+            if ($item['recommender_id']) {
+                $recommenderId = $item['recommender_id'];
+                $recommender = $this->_customerRepositoryInterface->getById($recommenderId);
+                $item['recommender_id'] = $recommender->getEmail();
```

Comment:
Co w sytuacji, gdy użytkownik wyśle rekomendacje, a następnie jego konto zostanie usunięte? W naszej tabeli taki użytkownik wciąż będzie, ale `$this->_customerRepositoryInterface->getById($recommenderId)` rzuci `\Magento\Framework\Exception\NoSuchEntityException`. Przydałoby się ten wyjątek wyłapać i obsłużyć, aby aplikacja wciąż w takim wypadku działała.